src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/UnnamedSymbolCompletionProvider.cs

```diff
@@ -0,0 +1,200 @@
+
+            AddUnnamedSymbols(context, position, semanticModel, recommendedSymbols.UnnamedSymbols, cancellationToken);
+        }
+
+        private void AddUnnamedSymbols(
+            CompletionContext context, int position, SemanticModel semanticModel, ImmutableArray<ISymbol> unnamedSymbols, CancellationToken cancellationToken)
+        {
+            // Add one 'this[]' entry for all the indexers this type may have.
+            AddIndexers(context, unnamedSymbols.WhereAsArray(s => s.IsIndexer()));
+
+            // Group all the related operators and add a single completion entry per group.
+            var operatorGroups = unnamedSymbols.WhereAsArray(s => s.IsUserDefinedOperator()).GroupBy(op => op.Name);
+            foreach (var opGroup in operatorGroups)
+                AddOperatorGroup(context, opGroup.Key, opGroup);
+
+            foreach (var symbol in unnamedSymbols)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                if (symbol.IsConversion())
+                    AddConversion(context, semanticModel, position, (IMethodSymbol)symbol);
+            }
+        }
+
+        internal override Task<CompletionChange> GetChangeAsync(
+            Document document,
+            CompletionItem item,
+            TextSpan completionListSpan,
+            char? commitKey,
+            bool disallowAddingImports,
+            CancellationToken cancellationToken)
+        {
+            var kind = item.Properties[KindName];
+            return kind switch
+            {
+                IndexerKindName => GetIndexerChangeAsync(document, item, cancellationToken),
+                OperatorKindName => GetOperatorChangeAsync(document, item, cancellationToken),
+                ConversionKindName => GetConversionChangeAsync(document, item, cancellationToken),
+                _ => throw ExceptionUtilities.UnexpectedValue(kind),
+            };
+        }
+
+        public override async Task<CompletionDescription?> GetDescriptionAsync(
+            Document document,
+            CompletionItem item,
+            CancellationToken cancellationToken)
+        {
+            var kind = item.Properties[KindName];
+            return kind switch
+            {
+                IndexerKindName => await GetIndexerDescriptionAsync(document, item, cancellationToken).ConfigureAwait(false),
+                OperatorKindName => await GetOperatorDescriptionAsync(document, item, cancellationToken).ConfigureAwait(false),
+                ConversionKindName => await GetConversionDescriptionAsync(document, item, cancellationToken).ConfigureAwait(false),
+                _ => throw ExceptionUtilities.UnexpectedValue(kind),
+            };
+        }
+
+        private static Task<CompletionChange> ReplaceTextAfterOperatorAsync(Document document, CompletionItem item, string text, CancellationToken cancellationToken)
```

Comment:
nits: those two methods are not called in this partial class, maybe they could be moved to the class where they are called